.husky/pre-commit

```diff
@@ -1,11 +1,49 @@
 #!/usr/bin/env sh
 
-lint-staged
-
-types_path="packages/calcite-components/src/components.d.ts"
+ensure_types_are_up_to_date() {
+  types_path="packages/calcite-components/src/components.d.ts"
 
-# make sure the types are always up to date
-if [ -n "$(git diff --name-only -- "$types_path")" ]; then
+  if [ -n "$(git diff --name-only -- "$types_path")" ]; then
     echo "Automatically staging changes to \"$types_path\""
     git add "$types_path"
-fi
+  fi
+}
+
+check_ui_icon_name_consistency() {
+  SVG_ICON_DIR="packages/calcite-ui-icons/icons"
+
+  # this pattern checks for `<iconName>-<size>.svg` or `<iconName>-<size>-f.svg` for filled icons
+  # where `<iconName>` is kebab-case, `<size>` is 16, 24, or 32
+  valid_pattern="^[a-z0-9-]+-(16|24|32)(-f)?\\.svg$"
+
+  # this pattern will check for invalid use of "-f-" anywhere except right before the size
+  invalid_pattern="-[a-z0-9]+-f-"
+
+  STAGED_FILES=$(git diff --cached --name-only --diff-filter=ACM)
+
+  for file in $STAGED_FILES; do
```

Comment:
```suggestion
  for file in "$STAGED_FILES"; do
```

Reply:
Quoting this one gives me the following error:

> Since you double quoted this, it will not word split, and the loop will only run once.